codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriterTest.java

```diff
@@ -0,0 +1,270 @@
+        when(ctx.executor()).thenReturn(ImmediateEventExecutor.INSTANCE);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        outbound.release();
+        expectedOutbound.release();
+        frameWriter.close();
+    }
+
+    @Test
+    public void writeHeaders() throws Exception {
+        int streamId = 1;
+        Http2Headers headers = new DefaultHttp2Headers()
+                .method("GET").path("/").authority("foo.com").scheme("https");
+
+        frameWriter.writeHeaders(ctx, streamId, headers, 0, true, promise);
+
+        byte[] expectedPayload = headerPayload(streamId, headers);
+        byte[] expectedFrameBytes = {
+                (byte) 0x00, (byte) 0x00, (byte) 0x0a, // payload length = 10
+                (byte) 0x01, // payload type = 1
+                (byte) 0x05, // flags = (0x01 | 0x04)
+                (byte) 0x00, (byte) 0x00, (byte) 0x00, (byte) 0x01 // stream id = 1
+        };
+        expectedOutbound = Unpooled.copiedBuffer(expectedFrameBytes, expectedPayload);
+        assertEquals(expectedOutbound, outbound);
+    }
+
+    @Test
+    public void writeHeadersWithPadding() throws Exception {
+        int streamId = 1;
+        Http2Headers headers = new DefaultHttp2Headers()
+                .method("GET").path("/").authority("foo.com").scheme("https");
+
+        frameWriter.writeHeaders(ctx, streamId, headers, 5, true, promise);
+
+        byte[] expectedPayload = headerPayload(streamId, headers, (byte) 4);
+        byte[] expectedFrameBytes = {
+                (byte) 0x00, (byte) 0x00, (byte) 0x0f, // payload length = 16
+                (byte) 0x01, // payload type = 1
+                (byte) 0x0d, // flags = (0x01 | 0x04 | 0x08)
+                (byte) 0x00, (byte) 0x00, (byte) 0x00, (byte) 0x01 // stream id = 1
+        };
+        expectedOutbound = Unpooled.copiedBuffer(expectedFrameBytes, expectedPayload);
+        assertEquals(expectedOutbound, outbound);
+    }
+
+    @Test
+    public void writeHeadersNotEndStream() throws Exception {
+        int streamId = 1;
+        Http2Headers headers = new DefaultHttp2Headers()
+                .method("GET").path("/").authority("foo.com").scheme("https");
+
+        frameWriter.writeHeaders(ctx, streamId, headers, 0, false, promise);
+
+        byte[] expectedPayload = headerPayload(streamId, headers);
+        byte[] expectedFrameBytes = {
+                (byte) 0x00, (byte) 0x00, (byte) 0x0a, // payload length = 10
+                (byte) 0x01, // payload type = 1
+                (byte) 0x04, // flags = 0x04
+                (byte) 0x00, (byte) 0x00, (byte) 0x00, (byte) 0x01 // stream id = 1
+        };
+        ByteBuf expectedOutbound = Unpooled.copiedBuffer(expectedFrameBytes, expectedPayload);
+        assertEquals(expectedOutbound, outbound);
+    }
+
+    /**
+     * Test large headers that exceed {@link DefaultHttp2FrameWriter#maxFrameSize}
+     * the remains headers will send by CONTINUATION frame
```

Comment:
`the remains headers will send by` -> `the remaining headers will be sent in a CONTINUATION frame`